cql-engine/src/main/java/org/opencds/cqf/cql/elm/execution/RetrieveEvaluator.java

```diff
@@ -31,23 +31,23 @@ public Object evaluate(Context context) {
             else {
                 Object codesResult = this.getCodes().evaluate(context);
                 if (codesResult instanceof String) {
-                    ArrayList codesList = new ArrayList<>();
+                    List<Code> codesList = new ArrayList<>();
                     codesList.add(new Code().withCode((String)codesResult));
                     codes = codesList;
                 }
                 else if (codesResult instanceof Code) {
-                    ArrayList<Code> codesList = new ArrayList<>();
+                    List<Code> codesList = new ArrayList<>();
                     codesList.add((Code)codesResult);
                     codes = codesList;
                 }
                 else if (codesResult instanceof Concept) {
-                    ArrayList<Code> codesList = new ArrayList<>();
+                    List<Code> codesList = new ArrayList<>();
                     for (Code conceptCode : ((Concept)codesResult).getCodes()) {
                         codesList.add(conceptCode);
                     }
                     codes = codesList;
                 }
-                else {
+                else if (codesResult instanceof Iterable) {
```

Comment:
I think it would be better to let the cast throw, otherwise this will turn in to a NullPointerException when it gets to the data access layer (or worse, an unfiltered retrieve with no error).